lib/node_modules/@stdlib/array/base/count-if/docs/types/index.d.ts

```diff
@@ -0,0 +1,46 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { Collection } from '@stdlib/types/array';
+
+/**
+* Counts the number of truthy values in an array.
+*
+* @param x - input array
+* @param predicate - testing function
+* @param thisArg - function context
+* @returns number of values for which the provided function evaluates to true
+*
+* @example
+* var x = [ 0, 1, 0, 1 ];
+* var predicate = function( v ) {
+*     return v > this;
+* }
+* var n = indexed( x, predicate );
+* // returns 2
+*/
+declare function countIf( x: Collection, predicate: ( ...args: Array<any> ) => boolean, thisArg?: any ): number;
```

Comment:
We can improve the type definitions for this functions quite a bit, I think. 
You may want to have a look, @Ut-the-pro, at say the [ones for `every-by`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/array/base/every-by/docs/types/index.d.ts) for some inspiration on how we can type the `predicate` function and the `thisArg` parameter.